force-app/main/default/classes/ServiceScheduleService.cls

```diff
@@ -43,6 +43,45 @@ public with sharing class ServiceScheduleService {
         return serviceSelector.getExistingParticipantContactIdsByScheduleId(scheduleId);
     }
 
+    public Boolean checkAddParticipantsFieldPermissions() {
```

Comment:
What do you think about dropping Field in the name? Since technically it depends on object access as well?